server/athenian/api/align/goals/measure.py

```diff
@@ -187,26 +189,31 @@ def __init__(
         )
 
     @property
-    def request(self) -> TeamMetricsRequest:
+    def requests(self) -> Iterable[TeamMetricsRequest]:
         """Get the request for calculate_team_metrics to compute the metrics for the goal."""
-        return self._request
+        return self._requests
 
     def build_goal_tree(self, metric_values: TeamMetricsResult) -> GoalTree:
         """Build the GoalTree combining the teams structure and the metric values."""
-        intervals = self._request.time_intervals
-        metric = self._request.metrics[0]
-
-        # for initial and current metrics the interval is a pair,
-        # we produce and need only one value from TeamMetricsResult
-        initial_metrics_base = metric_values[intervals[0]][metric]
-        initial_metrics = {k: values[0] for k, values in initial_metrics_base.items()}
-        current_metrics_values = metric_values[intervals[1]][metric]
-        current_metrics = {k: values[0] for k, values in current_metrics_values.items()}
+        intervals = self._requests[0].time_intervals  # the same in every requests
 
+        initial_metrics_base = metric_values[intervals[0]]
+        initial_metrics = {}
+        current_metrics_base = metric_values[intervals[1]]
+        current_metrics = {}
         if self._timeseries_spec is None:
-            series = None
+            series: Optional[dict] = None
         else:
-            series = metric_values[self._timeseries_spec.intervals][metric]
+            series = {}
+        # parse metric_values so that it can be more easily consumed by generator
+        # for each team read the value for the metric with the right parameters
+        for t_id_g_id, params in self._metrics_w_params_by_team.items():
+            initial_metrics[t_id_g_id] = initial_metrics_base[params][t_id_g_id][0]
+            current_metrics[t_id_g_id] = current_metrics_base[params][t_id_g_id][0]
+            if series is not None and self._timeseries_spec is not None:
+                series[t_id_g_id] = metric_values[self._timeseries_spec.intervals][params][
+                    t_id_g_id
+                ]
```

Comment:
```suggestion
                # fmt: off
                series[t_id_g_id] = (
                    metric_values[self._timeseries_spec.intervals][params][t_id_g_id]
                )
                # fmt: on
```
the original format is too ugly